source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -216,24 +235,19 @@ void DnsCacheImpl::onReResolve(const std::string& host) {
 
   // Functions like this one that modify primary_hosts_ are only called in the main thread so we
   // know it is safe to use the PrimaryHostInfo pointers outside of the lock.
```

Comment:
should we move this comment to function definition and/or add a thread assert in the function?

Reply:
Yeah sorry forgot about this will do.